### PR TITLE
executor, plan: union scan reuse pk when it's handle.

### PR DESCRIPTION
**Pull request by winoros**
Reuse pk when it's handle.
PTAL @hanfei1991 @lamxTyler

### CODE REVIEW

**Comment by shenli** [comment]
There is no test case in this PR.

**Comment by hanfei1991** [comment]
@shenli existing tests can cover this change

File: executor/builder.go [hunk 1]
```diff
@@ -370,7 +370,15 @@ func (b *executorBuilder) buildUnionScanExec(v *plan.PhysicalUnionScan) Executor
 	if b.err != nil {
 		return nil
 	}
-	us := &UnionScanExec{baseExecutor: newBaseExecutor(v.Schema(), b.ctx, src), needColHandle: v.NeedColHandle}
+	us := &UnionScanExec{baseExecutor: newBaseExecutor(v.Schema(), b.ctx, src)}
+	for _, cols := range v.Children()[0].Schema().TblID2Handle {
```
Comment (shenli) [inline executor/builder.go]: Add comment for this logic.

File: executor/union_scan.go [hunk 2]
```diff
@@ -91,11 +91,14 @@ type UnionScanExec struct {
...
-	columns       []*model.ColumnInfo
-	needColHandle bool
+	usedIndex  []int
+	desc       bool
+	conditions []expression.Expression
+	columns    []*model.ColumnInfo
+
+	// belowHandleIndex is the handle's position of the below scan plan.
+	belowHandleIndex int
+	handleColIsUsed  bool
```
Comment (shenli) [inline executor/union_scan.go]: Add comment for this.

**Comment by winoros** [comment]
comment updated, PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Comment by winoros** [comment]
@hanfei1991 @lamxTyler PTAL

File: executor/builder.go [hunk 3]
```diff
@@ -370,7 +370,19 @@ func (b *executorBuilder) buildUnionScanExec(v *plan.PhysicalUnionScan) Executor
 	if b.err != nil {
 		return nil
 	}
-	us := &UnionScanExec{baseExecutor: newBaseExecutor(v.Schema(), b.ctx, src), needColHandle: v.NeedColHandle}
+	us := &UnionScanExec{baseExecutor: newBaseExecutor(v.Schema(), b.ctx, src)}
+	// Get the handle column index of the below plan.
+	// We can guarantee that there's must be only one col in the map.
```
Comment (zimulala) [inline executor/builder.go]: s/there's/there

**Comment by zimulala** [comment]
LGTM

**Pull request merged**